src/labone/nodetree/node.py

```diff
@@ -0,0 +1,863 @@
+    This way, the information does not have to be stored in the tree structure itself.
+    """
+    def __init__(self, path_to_value: dict['LabOneNodePath', T]):
+        self._path_to_value = path_to_value
+
+    def get(self, node: 'MetaNode') -> T:
+        return self.get_with_path(repr(node))
+
+    def get_with_segments(self, path_segemnts: list['NormalizedPathSegment']):
+        return self.get_with_path(join_path(*path_segemnts))
+
+    #@functools.cache # caching is a dict access anyway
+    def get_with_path(self, path: 'LabOneNodePath') -> T:
+        return self._path_to_value[path]
+
+
+class NodeTreeManager:
+    def __init__(self, session: KernelSession, path_to_info):
+        self.session = session
+        self._remembered_nodes = {}
+        self.path_to_info = path_to_info
+        self._paths: list[LabOneNodePath] = list(self.path_to_info.keys())
+        self.structure: NestedDict = build_structure(self._paths)
+        self.structure_info: TreeProp = TreeProp(self.path_to_info)
+
+    @classmethod
+    async def create(cls, session):
+        path_to_info = await session.list_nodes_info('*', ListNodesFlags.ABSOLUTE | ListNodesFlags.RECURSIVE
+                                                     | ListNodesFlags.LEAVES_ONLY)
+        return cls(session, path_to_info)
+
+    def get_nodetree(self, enter_prefix=True):
+        if not enter_prefix or len(self.structure.keys()) != 1:
+            return Node(self, (), self.structure)
+
+        common_prefix: tuple[NormalizedPathSegment] = (list(self.structure.keys())[0],)
+        return Node(self, common_prefix, self.structure[common_prefix[0]])
+
+    def raw_path_to_node(self, path: LabOneNodePath):
+        return self.path_segments_to_node(tuple(split_path(path)))
+
+    def path_segments_to_node(self,
+                              path_segments: t.Tuple[t.Union[str, int]],
+                              subtree_structure: NestedDict | None = None,
+                              path_aliases: dict[tuple[NormalizedPathSegment], tuple[
+                                  NormalizedPathSegment]] | None = None,
+                              use_cache=True
+                              ):
+        """
+        Caches Node-objects and enforces thereby a singleton pattern:
+        not more than one node-object per path_segments
+        """
+        # subtree_structure_id = frozendict(subtree_structure)   # todo take structure into account
+        # path_aliases_id = hash(tuple(sorted(path_aliases.items())))
+
+        if use_cache and path_segments in self._remembered_nodes.keys():
+            return self._remembered_nodes[path_segments]
+
+        result = Node(self, path_segments, subtree_structure, path_aliases)
+        if use_cache:
+            self._remembered_nodes[path_segments] = result
+        return result
+
+    @property
+    def paths(self):
+        return self._paths
+
+
+async def get_nodetree(session: KernelSession, enter_prefix=True):
```

Comment:
Move out of node module into helper and export through init.py

also, rename the it to `construct_nodetree`

Reply:
`enter_prefix` -> `hide_kernel_prefix`

Reply:
should I really move it to helper? It uses NodeTreeManager, which then would need to be imported into helper. This would almost result in a circular import

Reply:
mhm let's keep it in node.py then. But the export is still needed